### PR TITLE
Feature/login page

### PR DESCRIPTION
**Pull request by jakeharding**
Need to run npm install to update dependencies after merging this branch.

### CODE REVIEW

File: jukebox_dj/index/static/jukebox-dj/src/providers/user/user.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  and Angular DI.
+*/
+@Injectable()
+export class UserProvider {
+  private djsUrl = "/api/dev/djs";
+
+  constructor(public http: HttpClient) {}
+
+  get (): Observable<User> {
+    return this.http.get(`${this.djsUrl}/me`);
```
Comment (tmart2322) [inline jukebox_dj/index/static/jukebox-dj/src/providers/user/user.ts]: Where does the /me come from in the http request?

File: jukebox_dj/songs/views.py [hunk 2]
```diff
@@ -82,33 +84,51 @@ class Meta:
...
+        cookie = SongRequestCookie.objects.get(uuid=request.data.get('cookie'))
+        error_message = None
+        one_hour_ago = datetime.utcnow() - timedelta(hours=1)
         if SongRequest.objects.filter(
-                        Q(song__uuid=song_uuid) &
-                        Q(created_at__gte=datetime.datetime.utcnow() - datetime.timedelta(hours=1))).exists():
+                        Q(song=song) &
+                        Q(created_at__gte=one_hour_ago)).exists():
+            error_message = 'A request for %s has recently been made. Please try again soon.' % song.title
+        elif SongRequest.objects.filter(
```
Comment (tmart2322) [inline jukebox_dj/songs/views.py]: Awesome that the filter for 5 an hour is put in!

File: jukebox_dj/index/static/jukebox-dj/src/components/jukebox-header/jukebox-header.scss [hunk 3]
```diff
@@ -6,23 +6,28 @@ jukebox-header {
...
   }
 
-  @media (min-width: map-get($grid-breakpoints, lg) + 1) {
+  @media (min-width: map-get($grid-breakpoints, sm) + 1) {
     .hidden-md-up {
       display: none !important;
     }
   }
+
+  .back-button {
```
[the remark this answers is not in the document]
Comment (jakeharding) [reply jukebox_dj/index/static/jukebox-dj/src/components/jukebox-header/jukebox-header.scss]: This hides the back button without removing it from the DOM allowing the Jukebox DJ heading to be centered on small screens and prevent elements from moving when it is removed or added. It is a hacky solution and I like having the back button available but it shouldn't affect other elements.

File: jukebox_dj/index/static/jukebox-dj/src/pages/requester/requester.ts [hunk 4]
```diff
@@ -66,13 +59,36 @@ export class RequesterPage {
       this.event = event;
     });
 
-    // TODO Index page may get the event and pass event data to this page. Add condition when index page is ready.
     this.scrollCallback = this.getEventSongs.bind(this);
   }
 
+  filterSongs() {
```
[the remark this answers is not in the document]
Comment (jakeharding) [reply jukebox_dj/index/static/jukebox-dj/src/pages/requester/requester.ts]: We want a request to be made when the user types so the search is refreshed every time. The string has to be greater than two characters though. No timer needed.

File: jukebox_dj/index/static/jukebox-dj/src/providers/user/user.ts (hunk 1 above)
Comment (jakeharding) [reply jukebox_dj/index/static/jukebox-dj/src/providers/user/user.ts]: It gets the current logged in users information.

File: jukebox_dj/index/static/jukebox-dj/src/pages/login/login.html [hunk 5]
```diff
@@ -0,0 +1,37 @@
+<!--
+  Generated template for the LoginPage page.
+
+  See http://ionicframework.com/docs/components/#navigation for more info on
+  Ionic pages and navigation.
+-->
+
```
[the remark this answers is not in the document]
Comment (jakeharding) [reply jukebox_dj/index/static/jukebox-dj/src/pages/login/login.html]: Good idea.

**Pull request merged**